3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -297,25 +353,65 @@ object EnrichmentManager {
       }
     }
 
+    // Parse the page URI's querystring
+    val pageQsMap = pageUri match {
+      case Success(Some(u)) => CU.extractQuerystring(u, raw.source.encoding).map(_.some)
+      case _ => Success(None)
+    }
+
     // Marketing attribution
-    val campaign = pageUri.fold(
-      e => unitSuccessNel, // No fields updated
-      uri => uri match {
-        case Some(u) =>
-          registry.getCampaignAttributionEnrichment match {
-            case Some(ce) =>
-              ce.extractMarketingFields(u, raw.source.encoding).flatMap(cmp => {
-                event.mkt_medium = cmp.medium.orNull
-                event.mkt_source = cmp.source.orNull
-                event.mkt_term = cmp.term.orNull
-                event.mkt_content = cmp.content.orNull
-                event.mkt_campaign = cmp.campaign.orNull
-                cmp.success
-                })
-            case None => unitSuccessNel
+    val campaign = pageQsMap match {
+      case Success(Some(qsMap)) => registry.getCampaignAttributionEnrichment match {
+        case Some(ce) =>
+          ce.extractMarketingFields(qsMap).flatMap(cmp => {
+            event.mkt_medium = cmp.medium.orNull
+            event.mkt_source = cmp.source.orNull
+            event.mkt_term = cmp.term.orNull
+            event.mkt_content = cmp.content.orNull
+            event.mkt_campaign = cmp.campaign.orNull
+            event.mkt_clickid = cmp.clickId.orNull
+            event.mkt_network = cmp.network.orNull
+            cmp.success
+          })
+        case None => unitSuccessNel
+      }
+      case _ => unitSuccessNel
+    }
+
+    // Cross-domain tracking
+    val crossDomain = pageQsMap match {
```

Comment:
I agree, it's too complicated for the EnrichmentManager. I will move it into enrichments/web/PageEnrichments.scala.